### PR TITLE
[Metricbeat][ Add option to remove fields from comparison in mocked tests and mage target

### PR DESCRIPTION
**Pull request by sayden**
Added 3 things:
- Config option `remove_fields_from_comparison` to remove fields from the JSON generated and expected files to byte-to-byte comparison when asserting tests.
- Added a README.md file to the folder where the code is located showing current options, flags and use.
- Added a mage target called `mockedTests` to avoid navigating to `beats/metricbeat/mb/testing/data` to run tests.

The PR can be considered a WIP but some overlook might be nice @ruflin to know if you like the current heading and altitude :wink:

### CODE REVIEW

File: metricbeat/magefile.go [hunk 1]
```diff
@@ -93,6 +94,27 @@ func Update() error {
 	return sh.Run("make", "update")
 }
 
+// MockedTests runs the HTTP tests using the mocked data inside each {module}/{metricset}/testdata folder.
+// Use MODULE={module_name} to run only mocked tests with a single module
+// Use GENERATE=true or GENERATE=1 to regenerate JSON files
+func MockedTests() error {
+	goTestDefaultArgs := mage.DefaultGoTestUnitArgs()
+
+	goTestDefaultArgs.ExtraFlags = []string{"github.com/elastic/beats/metricbeat/mb/testing/data/."}
```
Comment (sayden) [inline metricbeat/magefile.go]: There's a `Package` field in `GoTestArgs` type but the order of the executed contents isn't correct when using it. By placing everything as `ExtraFlag` I could control the order of the flags and package run.

File: metricbeat/mb/testing/data/data_test.go [hunk 2]
```diff
@@ -211,7 +212,36 @@ func runTest(t *testing.T, file string, module, metricSetName string, config Con
...
+		}
+
+		for eventIndex := range expectedMap {
+			if err := expectedMap[eventIndex].Delete(fieldToRemove); err != nil {
+				t.Fatal(err)
+			}
+		}
+	}
+
+	output, err := json.Marshal(&data)
```
Comment (sayden) [inline metricbeat/mb/testing/data/data_test.go]: @ruflin I modified this from `MarshalIndent` to just `Marshal`. I understood that the idea of using indent was to write into the `data.json` file in pretty format but for comparison it wasn't nice when it failed as the console was escaping newlines like here
image

File: metricbeat/mb/testing/data/example.yml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+type: http
```
Comment (sayden) [inline metricbeat/mb/testing/data/example.yml]: Added some example config file for full reference :)

File: metricbeat/mb/testing/data/data_test.go (hunk 2 above)
Comment (ruflin) [reply metricbeat/mb/testing/data/data_test.go]: No strong preference as long as the files written to disk are indented. 

For the screenshot you posted: It shows the error pretty well I would say.

File: metricbeat/magefile.go (hunk 1 above)
Comment (ruflin) [reply metricbeat/magefile.go]: Probably @andrewkroh knows best here.

File: metricbeat/mb/testing/data/README.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
+## Event generation testing framework
+
+### Introduction
+
+#### What it is
+Many modules do requests to an HTTP endpoint to fetch metrics data. Then they do data manipulation and enriching of this data before sending it to Elasticsearch.
+
+What we wanted to achieve is to mock the HTTP responses from the modules into a generic server that will serve those responses. So, for each tested metricset, an HTTP server in port 5555 is launched with the mocked response from a fixed module version to respond to the HTTP requests, once the test is done, it's shut down.. This way we isolate the manipulation of data in some tests and reduce lot of boilerplate we had in many modules.
```
Comment (ruflin) [inline metricbeat/mb/testing/data/README.md]: It's not correct that the server is launched on 5555, it's a random port selected but we hardcode 5555 in the final event.

**Comment by sayden** [comment]
Closed this by mistake, I cannot reopen it now :nerd_face: 

PR has been split in 3:
* README change https://github.com/elastic/beats/pull/11714
* Mage target https://github.com/elastic/beats/pull/11715 
* Mocked tests new config option https://github.com/elastic/beats/pull/11716

**Pull request closed without merging**